src/kvstore/RocksEngine.cpp

```diff
@@ -232,8 +245,11 @@ nebula::cpp2::ErrorCode RocksEngine::rangeWithPrefix(const std::string& start,
                                                      const std::string& prefix,
                                                      std::unique_ptr<KVIterator>* storageIter) {
   rocksdb::ReadOptions options;
-  // prefix_same_as_start is false by default
-  options.total_order_seek = FLAGS_enable_rocksdb_prefix_filtering;
+  if (!isPlainTable_) {
+    options.total_order_seek = FLAGS_enable_rocksdb_prefix_filtering;
+  } else {
+    options.prefix_same_as_start = true;
```

Comment:
PlainTable must use prefix-based Seek, so we must prefix_same_as_start = true;
And for BlockBasedTable, if we have prefix extractor, we just use total_order_seek for simplicity, otherwise we need to judge whether the parameter is longer than extractor length (the way we use in prefix function)